backend/src/services/external-migration/external-migration-service.ts

```diff
@@ -0,0 +1,185 @@
+import slugify from "@sindresorhus/slugify";
+
+import { OrgMembershipRole, ProjectMembershipRole, SecretType } from "@app/db/schemas";
+import { BadRequestError } from "@app/lib/errors";
+import { logger } from "@app/lib/logger";
+import { alphaNumericNanoId } from "@app/lib/nanoid";
+
+import { TOrgServiceFactory } from "../org/org-service";
+import { TProjectServiceFactory } from "../project/project-service";
+import { TProjectEnvServiceFactory } from "../project-env/project-env-service";
+import { TSecretServiceFactory } from "../secret/secret-service";
+import { decryptEnvKeyData, parseEnvKeyData } from "./external-migration-fns";
+import { TImportEnvKeyDataCreate, TImportInfisicalDataCreate } from "./external-migration-types";
+
+type TExternalMigrationServiceFactoryDep = {
+  projectService: TProjectServiceFactory;
+  orgService: TOrgServiceFactory;
+  projectEnvService: TProjectEnvServiceFactory;
+  secretService: TSecretServiceFactory;
+};
+
+export type TExternalMigrationServiceFactory = ReturnType<typeof externalMigrationServiceFactory>;
+
+export const externalMigrationServiceFactory = ({
+  projectService,
+  orgService,
+  projectEnvService,
+  secretService
+}: TExternalMigrationServiceFactoryDep) => {
+  const importInfisicalData = async ({
+    data,
+    actor,
+    actorId,
+    actorOrgId,
+    actorAuthMethod
+  }: TImportInfisicalDataCreate) => {
+    // Import data to infisical
+    if (!data || !data.projects) {
+      logger.error("No projects found in data");
+      return {
```

Comment:
Throw a Bad Request error instead with the error message "No projects found in data"